### ISSUE TITLE
Possible incorrect Network usage in augmentLabels

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#54381]
https://github.com/istio/istio/blob/a26b3a14451971f3acde32e3c57eba0213020328/pilot/pkg/serviceregistry/kube/controller/controller.go#L1132
 It seems like it should take into a `account proxy.Metadata.Network` and/or use `c.Network()`? 

`c.network` == istio-system label while `c.Network()` can read from pod label, system namespace, or meshNetworks.

I just stumbled upon this doing something else, so not directly impacted and haven't heard any real user reports. These days I think the vast majority of users are using system namespace labels, which makes it _probably_ not a concern in practice.

fyi @ramaraochavali @hzxuzhonghu

**Comment by hzxuzhonghu** [comment]
c.Network() is used by endpoint builder, it may rely on the labels set by` labelutil.AugmentLabels`

proxy.Metadata.Network is from `ISTIO_META_NETWORK`, we set it into proxy.Labels in `setTopologyLabels`

So we prioritize ISTIO_META_NETWORK env over pod label.

**Comment by howardjohn** [comment]
it's currently using c.network though which is exclusively from the system namespace

**Comment by hzxuzhonghu** [comment]
I read the opposite, c.network is also referring system ns label.

```
// setNetworkFromNamespace sets network got from system namespace, returns whether it has changed
func (n *networkManager) setNetworkFromNamespace(ns *v1.Namespace) bool {
	nw := ns.Labels[label.TopologyNetwork.Name]
	n.Lock()
	defer n.Unlock()
	oldDefaultNetwork := n.network
	n.network = network.ID(nw)
	return oldDefaultNetwork != n.network
}
```

**Comment by howardjohn** [comment]
yeah that's what I mean - it's *only* considering the namespace label. shouldn't it also include things like MeshNetworks

**Comment by hzxuzhonghu** [comment]
Good catch `MeshNetworks`  is not respected here, one reason should be that when we implemented system ns. `meshNetworks`  is to be deprecated?
